earthpy/spatial.py

```diff
@@ -266,10 +274,10 @@ def bytescale(data, cmin=None, cmax=None, high=255, low=0):
     if high < low:
         raise ValueError("`high` should be greater than or equal to `low`.")
 
-    if (cmin is None) or (cmin < data.min()):
+    if cmin is None:
```

Comment:
This line reverts back to the original scipy implementation. The change for these conditional statements was necessary for the scipy examples to provide the expected results, and did not break any of our existing unit tests. I looked at the pull request (https://github.com/earthlab/earthpy/pull/145) where we added `or (cmin < data.min())` and `or (cmax > data.max())` and couldn't tell why the `or` statements were added. 

It would be great if @joemcglinchy could take a look at this change and make sure this change doesn't break something (and if it does, we should add a unit test to catch that edge case). But, the existing earthpy bytescale function did not generate the expected output for the third example in the bytescale docstring which was taken from the original scipy docstring - it was scaling values to have range (0, 255) whereas the expected output from the original scipy version has range (3, 91). 

My assumption here is that we want this function to behave like scipy's bytescale function, but if that's wrong, let me know!

Reply:
based on their documentation i would expect the third example to have output with range (0,255) the same as the first example, as opposed to directly converting the input array to uint8 by way of rounding. What their example seems to is say if the data is already in the range (0,255), ie is of type uint8, then do no scaling regardless of how the cmin/cmax parameters are specified. Currently, we are saying that if the cmin or cmax is less than or greater than the data.min() or data.max(), then use the data.min or data.max(). I think that is correct, but the only difference is we are still scaling the data in that case to the values specified by `low` and `high`. In other words, we are always stretching the data to be between 0-255 even if the data is already in that range. We could add a conditional that says if the data is already in that range, then just return the data with an `.astype('uint8')`, which would be appropriate.

Reply:
Cool - thanks @joemcglinchy! I agree that based on their documentation I would expect their third example to generate a range of values from (0, 255), rather than the original data range. 

So, we need to decide whether to stick with the original scipy behavior, which is: 

```python
>>> img = np.array([[ 91.06794177,   3.39058326,  84.4221549 ],
...                 [ 73.88003259,  80.91433048,   4.88878881],
...                 [ 51.53875334,  34.45808177,  27.5873488 ]])

>>> bytescale(img, cmin=0, cmax=255)
array([[91,  3, 84],
       [74, 81,  5],
       [52, 34, 28]], dtype=uint8)
```

or whether we want to rescale the values to have range (0, 255), even when the original data range is already contained within that range (the behavior after #145, and incidentally also the behavior of `gdal_translate`):

```python
>>> img = np.array([[ 91.06794177,   3.39058326,  84.4221549 ],
...                 [ 73.88003259,  80.91433048,   4.88878881],
...                 [ 51.53875334,  34.45808177,  27.5873488 ]])

>>> bytescale(img, cmin=0, cmax=255)
array([[255,  0, 237],
       [206, 226,  4],
       [141, 91, 71]], dtype=uint8)
```

I would advocate for the behavior in the second example, rather than the original scipy behavior, because it is more consistent with the documentation, and is identical to `gdal_translate` byte scaling. Does that sound reasonable @lwasser? If so, I will change the example in the docstring.

Reply:
i also advocate for parallelling `gdal_translate`. it makes the most sense to stretch the values between 0 and 255 as that is what a user would expect when plotting!! Great catch and thank you so much!!

Reply:
Cool - thanks again @joemcglinchy and @lwasser! Commit a11479a956403f7d1e8e9c3dfd1a3979e092edb9 makes the bytescale function act like we want.